File: server/api/schemas/loyalty.py
from typing import List, Optional

from pydantic import BaseModel


class LoyaltyResponse(BaseModel):
    id: int
    name: str
    discount: int
    description: Optional[str] = None
    requirements: Optional[dict] = None

    class Config:
        from_attributes = True


class LoyaltyDetailsResponse(BaseModel):
    id: int
    name: str
    discount: int
    description: Optional[str] = None
    requirements: str
    benefits: List[str]

    class Config:
        from_attributes = True

File: client/src/types/loyalty.ts
export interface Loyalty {
  id: number
  name: string
  discount: number
}

export interface LoyaltyResponse {
  id: number
  name: string
  discount: number
  description: string | null
  requirements: {
    text: string
    benefits: string[]
  } | null
}

export interface LoyaltyDetails extends Loyalty {
  description: string
  requirements: string
  benefits: string[]
}
